src/components/IconButton.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <div class="flex flex-col items-center w-fit">
+    <Button
+      :variant
+      :unstyled="true"
+      :class="classes"
+      :icon="'pi ' + icon"
+      @click="$emit('click')"
+      :pt="{
+        icon: {
+          class: iconSize,
+        },
+      }"
+    />
+    <p>{{ label }}</p>
+  </div>
+</template>
+<script setup lang="ts">
+import { Button } from 'primevue';
+import { computed } from 'vue';
+
+// CONSTANTS
+
+const iconMap = {
+  sun: 'pi-sun',
+  moon: 'pi-moon',
+  reset: 'pi-replay',
+  settings: 'pi-cog',
+};
+
+const iconSizeMap = {
+  small: 'text-sm!',
+  medium: 'text-md!',
+  large: 'text-lg!',
+  extraLarge: 'text-xl!',
+};
+
+// TYPES
+
+type IconButtonEmits = {
+  (e: 'click'): void;
+};
+
+type IconButtonProps = {
+  label?: string;
+  icon: keyof typeof iconMap;
+  size?: keyof typeof iconSizeMap;
+  variant?: 'transparent' | 'translucent' | 'primary';
+};
+
+// LOGIC
+
+defineEmits<IconButtonEmits>();
+
+const props = withDefaults(defineProps<IconButtonProps>(), {
+  variant: 'primary',
+  size: 'medium',
+});
+
+const iconSize = iconSizeMap[props.size];
+
+const icon = iconMap[props.icon];
```

Comment:
```suggestion
const iconSize = computed(() => iconSizeMap[props.size]);

const icon = computed(() => iconMap[props.icon]);
```

Die Reaktivität geht hier verloren, da du iconSize und icon als normale Variablen definiert hast. 
Damit das nicht passiert, würde ich empfehlen, beide als computed Properties zu definieren.
Alternativ könntest du auch einen watcher auf die Props setzen und die beiden Variablen als ref definieren, um sie bei Änderungen von außen zu aktualisieren.
Das wäre aber eher unschön und mit mehr Code verbunden.
Diese Lösung würde sich nur lohnen, wenn du die Werte innerhalb der Komponente selbst verändern möchtest, was in deinem Fall aber nicht notwendig ist.